src/LSST/control/OutputPortBits.h

```diff
@@ -0,0 +1,115 @@
+/*
+ *  This file is part of LSST M2 support system package.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_M2CELLCPP_CONTROL_OUTPUTPORTBITS_H
+#define LSST_M2CELLCPP_CONTROL_OUTPUTPORTBITS_H
+
+// System headers
+#include <functional>
+#include <map>
+#include <memory>
+
+// Project headers
+
+namespace LSST {
+namespace m2cellcpp {
+namespace control {
+
+/// Class representation of the output port bits as seen in create_output_port_bits_masks.vi.
+/// This class contains enums for the position of all bits for the output port.
+/// Unit tests in test_SimCore.cpp.
+class OutputPortBits {
+public:
+    using Ptr = std::shared_ptr<OutputPortBits>;
+
+    /// Create a OutputPortBits with a blank _bitmap
+    OutputPortBits() = default;
+
+    /// Create a OutputPortBits with a `_bitmap` set to `bitmap`.
+    OutputPortBits(uint8_t bitmap) : _bitmap(bitmap) {}
+
+    OutputPortBits(OutputPortBits const&) = default;
+    OutputPortBits& operator=(OutputPortBits const&) = default;
+    virtual ~OutputPortBits() = default;
+
+    /// These enums should match those in create_output_port_bits_masks.vi
+    /// Changing the values of these enums will likely cause hardware problems.
+    enum {
+        ILC_MOTOR_POWER_ON = 0,    ///< “ILC Motor Power On” = 0
```

Comment:
The motor power here is for the actuators/drives instead of ILC. The power of ILC comes from the communication power in `ILC_COMM_POWER_ON=1`.

Therefore, maybe `MOTOR_POWER_ON` or `ACTUATOR_MOTOR_POWER_ON` or `DRIVE_MOTOR_POWER_ON` or something similar.

Reply:
I'm trying to be consistent with the names as they come from the LabView code. It's awful that there are things named so badly, but matching up new C++ code with the old LabView vi's is difficult. By keeping the names consistent, I can at least find what inspired the C++ code in LabView.

At some point, when things work, it would be totally worth fixing these names.  I'll add a comment, but I'd really like to hold off on the name change for a while.

Reply:
I can probably drop the `ILC_` without it being too much of an issue here.